### PR TITLE
Refactor BootLoaderInstall

### PR DESCRIPTION
**Pull request by davidcassany**
This commit refactors BootLoaderInstall class to make it a proper
factory class. In addition type hints are added for the constructor
method.

Related to #1498

### CODE REVIEW

**Review by schaefi — APPROVED** [review]
This is great, but it will conflict with #1567 

Are you ok to wait for the merge until #1567 is merged and fix the conflict ? I assume the conflict will be very easy to resolve

**Comment by davidcassany** [comment]
Yes absolutely fine with me, no problem I'll fix the conflicts once #1567 is done.

**Comment by schaefi** [comment]
It will take a bit of time for me to fix the rest of the s390 stuff. I'll merge yours and fix the conflict in my branch.

**Pull request merged**
